src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -666,6 +687,27 @@ private static bool TryGetRpcExceptionFields(
             return true;
         }
 
+        private static bool TryExtractSerializedFailureDetailsFromException(string exception, out FailureDetails? details)
+        {
+            try
+            {
+                string serializedMessage = exception.Split('\n')[0];
```

Comment:
Two things:
1. `string.Split` can be a bit expensive. It's faster and less memory intense to do `IndexOf` and `Substring`.
2. Let's not attempt to parse the message unless we have some clue that it's JSON to avoid parsing things that obviously aren't JSON.

Suggestion:
```csharp
if (exception[0] != '{') // expecting JSON
{
    details = null;
    return false;
}

int newlineIndex = input.IndexOf('\n');
string failureDetailsJson = newlineIndex < 0 ? exception : exception.Substring(0, newlineIndex).Trim();
P.TaskFailureDetails? taskFailureDetails = JsonConvert.DeserializeObject<P.TaskFailureDetails>(failureDetailsJson);
// ...
```

Reply:
Implemented